gdal/frmts/hdf5/hdf5dataset.cpp

```diff
@@ -304,7 +313,25 @@ GDALDataset *HDF5Dataset::Open( GDALOpenInfo *poOpenInfo )
     poDS->SetDescription(poOpenInfo->pszFilename);
 
     // Try opening the dataset.
-    poDS->hHDF5 = H5Fopen(poOpenInfo->pszFilename, H5F_ACC_RDONLY, H5P_DEFAULT);
+    if (VSIStat(poOpenInfo->pszFilename, &statBuf) == 0 &&
+        VSI_ISREG(statBuf.st_mode))
+      {
+        poDS->hHDF5 = H5Fopen(poOpenInfo->pszFilename, H5F_ACC_RDONLY, H5P_DEFAULT);
+      }
+    else
+      {
+        // Not a regular file: ingest and use the HDF5 memory driver
+        if (!VSIIngestFile(nullptr, poOpenInfo->pszFilename, &poDS->fileBuffer, &filesz, -1))
```

Comment:
I'd like some limit based on the amount of RAM. For example CPLGetUsablePhysicalRAM() / 10 instead of -1

Reply:
I guess this is a philosophical argument. On heavily loaded systems, 10% of physical RAM will still be more than you would like to devote to a GDAL application. On lightly loaded systems, 10% will be far too low. Rather than have the library/application decide what a sensible level of memory usage is, why not just `malloc()` away as you like and control the application with `ulimit` and other OS resource control mechanisms? As long as GDAL sensibly handles memory allocation failure, then all should be fine.

Reply:
We can leave that as it for now. But on systems with virtual memory and disk swap, swap trashing tends to cause the system to nearly hangs before malloc() fails and very few people actually play with ulimit.